components/form.js

```diff
@@ -80,31 +104,73 @@ export default class Form extends React.Component {
   componentDidUpdate(prevProps, prevState) {
     localStorage.setItem("Form Fields", JSON.stringify(this.state.formFields));
     localStorage.setItem("Bearer Token", JSON.stringify(this.state.token));
+    window.addEventListener("online", this.isOnlineEvent);
   }
 
   // Handler html For when the submit button is pressed
   onSubmit = e => {
     e.preventDefault();
-    postFormData(this.state.formData, this.state.token.access_token);
+    let formDataArray = [];
+    if (navigator.onLine) {
+      postFormData(this.state.formData, this.state.token.access_token);
+    } else {
+      let formDataArray = JSON.parse(
+        localStorage.getItem("Failed Form Submission Data")
+      );
+      formDataArray.push(this.state.formData);
+
+      localStorage.setItem(
+        "Failed Form Submission Data",
+        JSON.stringify(formDataArray)
+      );
+    }
     this.setState({
       formData: {}
     });
   };
 
+  // Make state targets equal to the value of input fields
+  change = e => {
+    let data = this.state.tempData;
```

Comment:
mmm. This works, but I'm not convinced this is a good idea. The structure for the submission should be created on the onSubmit function, not every time the content changes. 
That way you avoid manipulating the structure every time there is a change and you also avoid the need for `tempData` entirely, keeping this cleaner